### ISSUE TITLE
[FEA] Create column with a range of values

### ISSUE THREAD

**Issue by revans2** [rapidsai/cudf#4325]
**Is your feature request related to a problem? Please describe.**
In Spark there is a function that assigns a unique id to each row. 

From the spark documentation.
```
monotonically_increasing_id:

A column that generates monotonically increasing 64-bit integers.

The generated ID is guaranteed to be monotonically increasing and unique, but not consecutive. The current implementation puts the partition ID in the upper 31 bits, and the record number within each partition in the lower 33 bits. The assumption is that the data frame has less than 1 billion partitions, and each partition has less than 8 billion records.
```

I can handle creating the partition ID and putting it in the upper 31 bits.  But I need a way to generate the record number. To do this I need something to create a column with values starting at N and going to N+num_rows. 

**Describe the solution you'd like**
A simple function that will create a column like this. 

Basically a counting iterator, materialized into a column
https://thrust.github.io/doc/classthrust_1_1counting__iterator.html

**Describe alternatives you've considered**
I could build this on the CPU and send it down, but that would be slower and I can see other situations where someone might want the row number as a value.

**Additional context**
Add any other context, code examples, or references to existing implementations about the feature request here.

**Comment by jrhemstad** [comment]
So basically a `cudf::sequence` function akin to a [`thrust::sequence`](https://thrust.github.io/doc/group__transformations_ga9d17b70466df6b9a57f94170935a03f8.html)? 

```
unique_ptr<column> sequence(size_type size, scalar const& init);
```

Generates `[ init, init+1, init+2, ... init + (size - 1)]`;

Only valid for numeric types.

**Comment by kkraus14** [comment]
We could absolutely use this on the Python side as well. Ideally we could have a stepsize as well to allow for something like generating: [1, 3, 5, 7, 9, ...] and also handle negative stepsizes to generate something like: [9, 7, 5, 3, 1, -1, -3, ...]

**Comment by nvdbaranec** [comment]
filling.hpp?

**Comment by kkraus14** [comment]
> Only valid for numeric types.

It would be great to support this for `TIMESTAMP` types as well, but given we don't have a `DURATION` type for the scalar yet it doesn't exactly make sense quite yet.

**Comment by jrhemstad** [comment]
> filling.hpp?

Makes sense to me.

**Comment by jrhemstad** [comment]
> Any obscure or weird reasons this should somehow support strings?

Nope, I don't think so. Numeric only.

**Comment by lmeyerov** [comment]
From the df world, there's `np.arange()` (which even the cudf docs use). All over our code :)

**Comment by kkraus14** [comment]
> From the df world, there's `np.arange()` (which even the cudf docs use). All over our code :)

@lmeyerov you can use `cupy.arange()` and it should "just work" zero copy 😄

**Comment by lmeyerov** [comment]
Ah awesome. Feels like this should be more upfront in cudf docs, feels like a *whoosh* moment. (I had searched the API docs on this just yesterday!)
